src/index.ts

```diff
@@ -99,24 +101,36 @@ export default class OpentracingExtension<TContext extends SpanContext>
     source: any,
     args: { [argName: string]: any },
     context: TContext,
-    info: GraphQLResolveInfo
+    info: ExtendedGraphQLResolveInfo
   ) {
     if (
+      // we don't trace the request
       !this.requestSpan ||
-      !this.shouldTraceFieldResolver(source, args, context, info)
+      // we should not trace this resolver
+      !this.shouldTraceFieldResolver(source, args, context, info) ||
+      // the previous resolver was not traced
+      (info.path && info.path.prev && !context.getSpanByPath(info.path.prev))
     ) {
       return;
     }
 
+    // indepotent method to add helpers to the first context available (which will be propagated by apollo)
+    addContextHelpers(context);
+
     const name = info.fieldName || "field";
-    const parentSpan = context.span ? context.span : this.requestSpan;
+    const parentSpan =
+      info.path && info.path.prev
+        ? context.getSpanByPath(info.path.prev)
+        : this.requestSpan;
 
     const span = this.localTracer.startSpan(name, {
       childOf: parentSpan || undefined
     });
 
-    // There is no nicer way to change the span in the context
-    context.span = span;
+    context.addSpan(span, info);
+
+    // expose to field
+    info.span = span;
```

Comment:
I will likely avoid this method anyway just for some extra future proofing... just in case the object ever get freeze'd or graphql decides to recycle info objects.

Reply:
How would you communicate out of this plugin into the field resolver then?

Reply:
hmmm I wasn't thinking about the plugin only resolver. Would `context.getSpanByPath` not work there? Also I am not opposed to adding it. I just would just not use it in my resolvers

Reply:
I personally would like to hide these context methods and have the user rely on the info part instead. Reason is I don't want anyone to overwrite any of this by accident

Reply:
`Object.defineProperty(obj, "span", { value: span, configurable: false, writable: false });` may also be useful. (You learn something new everyday).

Reply:
My thinking was this, but maybe a mix: https://twitter.com/PascalPrecht/status/1053198969247944704